lib/codegen/CreateTerm.cpp

```diff
@@ -651,6 +656,21 @@ llvm::Value *CreateTerm::createHook(KORECompositePattern *hookAtt, KOREComposite
     llvm::Value *first = (*this)(pattern->getArguments()[0].get()).first;
     llvm::Value *second = (*this)(pattern->getArguments()[1].get()).first;
     return llvm::BinaryOperator::Create(llvm::Instruction::URem, first, second, "hook_MINT_urem", CurrentBlock);
+  } else if (name == "KREFLECTION.kompiledDir") {
```

Comment:
This is not typically how we implement most hooks. The ones here are implemented this way out of necessity, but in general, we prefer to use the libraries in the `runtime` directory. You are looking to define a function named `hook_KREFLECTION_kompiledDir`